tests/distributed_mode/test_pipeline.py

```diff
@@ -63,3 +64,48 @@ def test_subpipeline_is_not_recreated_in_remote_workers():
     assert len(subpipelines) == 1
     assert len(documents) == 3
     assert len(nlp.pipeline[0].keys()) == 2
+
+
+def test_subpipeline_is_added_and_removed():
+
+    tagger = SimpleTagger(attribute="noun", lookups=["he", "she"], tag=True)
+
+    name = "my_tagger"
+    nlp.add_pipe(tagger, name=name)  # Add tagger to the subpipeline
+
+    # Assert nlp subpipeline has two components
+    assert len(nlp.pipeline) == 2  # Tokenizer is added by default
+
+    nlp.remove_pipe(name=name)  # Remove tagger from the subpipeline
+    # Assert subpipeline component is removed
+    assert len(nlp.pipeline) == 1
+
+
+def test_subpipeline_is_added_to_remote_worker():
+
+    james = sy.VirtualWorker(hook, id="james")
+
+    # Create a PySyft Strings and send it to remote worker
+    text_ptr = String("building syfertext").send(james)
+
+    # Add tagger with remote = True
+    tagger = SimpleTagger(attribute="noun", lookups=["syfertext"], tag=True)
+    nlp.add_pipe(tagger, name="noun_tagger", remote=True)
+
+    # Should return a DocPointer to the doc on bob's machine
+    doc = nlp(text_ptr)
+    assert isinstance(doc, DocPointer)
+
+    # assert only one document object on james's machine
+    documents = [v for v in james._objects.values() if isinstance(v, Doc)]
+    assert len(documents) == 1
+
+    # assert only one subpipeline object on james's machine
+    subpipeline = [v for v in james._objects.values() if isinstance(v, SubPipeline)]
+    assert len(subpipeline) == 1
```

Comment:
the name of the variable would be better if you call it  subpipelins (with an 's') since it is a list